### ISSUE TITLE
keyname should use occurrenceID not catalogNumber

### ISSUE THREAD

**Issue by laurarussell** [VertNet/dwc-indexer#40]
keyname should use occurrenceID not catlogNumber if occurrenceID is populated.  If occurrenceID is null, then keyname should use catalogNumber.

When keyname uses catalogNumber it creates issue for individuals with multiple occurrences.  Only one occurrence shows in the index/portal when there should be multiple.

MSB Mamm 142856 should have two occurrence records. 

occurrenceID: http://arctos.database.museum/guid/MSB:Mamm:142856?seid=674513
occurrenceID: http://arctos.database.museum/guid/MSB:Mamm:142856?seid=1884963

But only one record is returned in the portal: http://portal.vertnet.org/search?q=catalognumber:142856+collectioncode:Mammal+specimens+institutioncode:MSB

**Comment by tucotuco** [comment]
Currently the catalogNumber is used to create the id for the index if it exists, rather than the occurrenceID. Thus, if more than one record for the same dataset has the same catalogNumber, only the last one will remain in the index. We observe this with MSB Mammals, for example, which should have 24 catalogNumbers with multiple occurrences. 

Relevant code can be found at https://github.com/VertNet/dwc-indexer/blob/master/utils.py#L461

Switch the preference to construct the key from the occurrenceID, thus:
```
    occ_id = ''
    if occid is not None and len(occid) > 0:
        occ_id = '%s' % re.sub("\'",'',repr(occid))
    else:
        if catnum is not None and len(catnum) > 0:        
            occ_id = re.sub("\'",'',repr(slugify(catnum)))
        else:
            occ_id = 'noid'
    return occ_id
```

**Comment by tucotuco** [comment]
This is a big change with multiple consequences. The first time indexing is done after making this change, the entire corpus of the index will have to be updated. Migrators will also have to be updated to change set the value of the references field. Afterwards, we should check all the Search and API docs for examples that have to be changed as a result.

**Comment by tucotuco** [comment]
Check if there is anything in the webapp associated with the following that needs to be changed as a result:

https://github.com/VertNet/webapp/blob/master/www/js/app/models/detail.js#L277

**Comment by tucotuco** [comment]
Fixed in post-harvest-processor in commit 199ce7398ab2890ada701f712a7e6b6db4b431ae
